src/DacpacTool/VersionChecker.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Threading;
+using System.Net.Http.Json;
+using NuGet.Versioning;
+using System.IO;
+using System.Globalization;
+
+namespace MSBuild.Sdk.SqlProj.DacpacTool
+{
+
+    public class VersionChecker
+    {
+        private readonly IConsole _console;
+        private readonly IVersionProvider _versionProvider;
+
+        public VersionChecker(IConsole console, IVersionProvider versionProvider)
+        {
+            _versionProvider = versionProvider;
+            _console = console;
+        }
+
+        public async Task CheckForPackageUpdateAsync()
+        {
+            try
+            {
+                var timeOut = TimeSpan.FromSeconds(2);
+
+                using var cts = new CancellationTokenSource(timeOut);
+
+                var cacheFile = Path.Join(Path.GetTempPath(), "MSBuild.Sdk.SqlProj.tag-" + DateTime.Now.ToString("yyyy-MM-dd", CultureInfo.InvariantCulture) + ".txt");
+
+                NuGetVersion latestVersion = null;
+
+                if (File.Exists(cacheFile))
+                {
+                    var cache = await File.ReadAllTextAsync(cacheFile, cts.Token).ConfigureAwait(false);
+                    latestVersion = NuGetVersion.Parse(cache);
+                }
+                else
+                {
+                    using var httpClient = new System.Net.Http.HttpClient
+                    {
+                        DefaultRequestHeaders = { { "User-Agent", "MSBuild.Sdk.SqlProj" } },
+                        Timeout = timeOut,
+                    };
+
+                    var response = await httpClient.GetFromJsonAsync<Release>("https://api.github.com/repos/rr-wfm/MSBuild.Sdk.SqlProj/releases/latest").ConfigureAwait(false);
+                    if (response is null || response.tag_name is null || response.draft || response.prerelease)
```

Comment:
Hmm, if we publish a prerelease or draft release then we're not doing the version check at all? Does that make sense?

Reply:
I think that makes sense?

Reply:
A prerelease or draft would be something you would explicitly opt in to use, I think

Reply:
I have changed to always warn